#!/usr/bin/env python
PACKAGE = "simple_follower"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("linearfront_p", double_t, 0, "linearfront_p", 0.4, 0.0, 2.0)
gen.add("linearback_p", double_t, 0, "linearback_p", 0.5, 0.0, 2.0)

gen.add("angularleft_p", double_t, 0, "angularleft_p", 3.3, 0.0, 5.0)
gen.add("angularright_p", double_t, 0, "angularright_p", 3.0, 0.0, 5.0)

gen.add("d_param", double_t, 0, "d_param", 0.6, 0.0, 1.0)

gen.add("max_angular_speed", double_t, 0, "max_angular_speed", 0.5, 0.0, 1.0)
gen.add("min_angular_speed", double_t, 0, "min_angular_speed", -0.5, -1.0, 1.0)

gen.add("max_linear_speed", double_t, 0, "max_linear_speed", 0.2, 0.0, 1.0)
gen.add("min_linear_speed", double_t, 0, "min_linear_speed", -0.2, -1.0, 1.0)

gen.add("goal_x", double_t, 0, "goal_x", 0.6, 0.0, 1.0)
gen.add("goal_y", double_t, 0, "goal_y", 0.0, 0.0, 1.0)

exit(gen.generate(PACKAGE, "simple_follower", "arPID"))
